File: 2022 Training/Tópicos/Contest 2/2340.py


n = int(input())
ans = 0
max_ = 0

for i in range(n):
    aux_a = int(input())
    aux_b = int(input())
    resp = (aux_a ^ aux_b)
    if (max_ < resp):
        max_ = resp
        ans = i

print(ans+1)
print()

File: 2022 Training/Tópicos/Contest 2/2340.cpp
#include <bits/stdc++.h>

using namespace std;

int main () {
    int n;
    cin >> n;
    int max_ = 0;
    int ans;

    for (int i = 0; i < n; i++) {
        int aux_a, aux_b;
        cin >> aux_a >> aux_b;
        double resp = pow(aux_a, log10(aux_b));
        cout << resp << endl;
        if (resp > max_) {
            ans = i;
            max_ = resp;
        }
    }

    cout << ans+1 << endl;
    return 0;
}